### PR TITLE
Handle unit type variables

### PR DESCRIPTION
**Pull request by DanielElisenberg**
# HANDLE UNIT TYPE VARIABLES
Unit type variables based on central definition now has the correct dataType and needs to be transformed to "Long" if it is being pseudonymized.

### What's changed?
- Changed the transformer to handle dataType of pseudonymized variables
- Extracted the file handling logic from the transformer module as it should not have this responsibility
- Updated the unit tests

### What's next?
I am not in favor of unit testing our systems too much, but considering how much tricky logic exist in these pipeline steps, I want to work on adding a few test. As well as cleaning up existing test cases and adding docstrings to make the tests function better as documentation.

### CODE REVIEW

File: job_executor/worker/steps/dataset_transformer.py [hunk 1]
```diff
@@ -167,15 +167,20 @@ def _transform_variable(variable: dict, role: str, start: str, stop: str):
...
         "name": variable["shortName"],
         "label": _get_norwegian_text(variable["name"]),
-        "notPseudonym": (
-            "unitType" not in variable
-            or not variable["unitType"]["requiresPseudonymization"]
+        "notPseudonym": not_pseudonym,
+        "dataType": (
+            _transform_data_type(variable["dataType"])
+            if not_pseudonym
+            else "Long"
```
Comment (vaskalan) [inline job_executor/worker/steps/dataset_transformer.py:179]: Looks good!

**Review by vaskalan — APPROVED** [review]
Looks fine to me!

File: job_executor/worker/steps/dataset_transformer.py [hunk 2]
```diff
@@ -167,15 +164,20 @@ def _transform_variable(variable: dict, role: str, start: str, stop: str):
...
```
Comment (haraldsvik) [inline job_executor/worker/steps/dataset_transformer.py:179]: Great work!

File: tests/resources/worker/steps/transformer/input_data/__init__.py
Comment (haraldsvik) [inline tests/resources/worker/steps/transformer/input_data/__init__.py:1]: 💯

**Review by haraldsvik — APPROVED** [review]
Looks good!

**Pull request merged**